### ISSUE TITLE
Pksm transfer not setting friendship right

### ISSUE THREAD

**Issue by pokelector** [FlagBrew/PKSM#940]
Pksm is not using proper mechanics for friendship transfers. AdmiralFish asked me to put here as reminder.

**Comment by Mike007899** [comment]
While Fish may know whats going on, others do not. Please explain in detail how, why, and what happens with this issue. Thanks.

-Mew

**Comment by pokelector** [comment]
Well,
Gen7 should have the transfer like this:

OT's friendship is the same, and CT(current handler) should be reset to base.
Right now, CT is reset to 0 but OT is fine. 

For example on file A i hatch a Mimikyu. It has friendship 120. I transfer it to file B, so it will have CT/OT 70/120. Pksm makes it 0/120. As i said, OT friendship is good but CT is not.

Gen4 resets both CT and OT to base. So CT is right, but OT I think is wrong. Not that this matters because you can never trade it back so only CT matters. Since CT is done proprly gen 4 doesnt need to be fixed. Only gen 7 does.

Tldr:
Gen7 CT is set to 0 as opposed to base. OT friendship is fine.

**Comment by Admiral-Fish** [comment]
Relevant links

PK6/7: https://github.com/kwsch/PKHeX/blob/master/PKHeX.Core/PKM/Shared/_K6.cs#L116

PK5: https://github.com/kwsch/PKHeX/blob/master/PKHeX.Core/PKM/PK5.cs#L307

PK4: https://github.com/kwsch/PKHeX/blob/master/PKHeX.Core/PKM/PK4.cs#L339

**Comment by pokelector** [comment]
kwsch/PKHeX#2269
Same thing in PKHeX. Make of this what you will.

**Comment by pokelector** [comment]
So i guess this doesnt need to be fixed because pkhex does the same thing.
